### DISCUSSION TITLE
creating-a-custom-error-handler-middleware-function/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#96] [Announcements]
# creating-a-custom-error-handler-middleware-function/

In this post I show how to customise the ExceptionHandlerMiddleware to create custom responses when an error occurs in your middleware pipeline

https://andrewlock.net/creating-a-custom-error-handler-middleware-function/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 23 2020, 11:18, in reply to [Marco](https://disqus.com/by/mdissel/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/96#discussioncomment-6058152)</em>

---

<p>Unfortunately, not easily. Your best option is probably to use the standard re-execute capabilities, execute an MVC action, and use the built-in content-negotiation capabilities there :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at October 28 2020, 06:40, in reply to [Paulo Morgado](https://disqus.com/by/paulo_morgado/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/96#discussioncomment-6058173)</em>

---

<p>Sure, I mention that in the post 😉</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 02 2020, 04:16, in reply to [Sh P](https://disqus.com/by/disqus_1RxqNj9Mun/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/96#discussioncomment-6058181)</em>

---

<p>Unfortunately there's not an elegant way of doing that currently. It's on the aspnet team's radar as part of "project houdini" though - essentially allowing you to use ActionResults outside of MVC/Razor Pages. Until then you're stuck doing things manually I'm afraid</p>

[a comment by an automated account is not shown]
